### PR TITLE
add user log in, log out, profile update with remember

### PR DESCRIPTION
**Pull request by kian1213**
Hello sir, please review mu pull request. I recreated a new branch, I get messed up in my last branch, I committed more than 15 files. 
I commit first the log in with remember user. Ill do the account activation and reset password in another branch. Thanks po

### CODE REVIEW

**Comment by codegracie** [comment]
Ok.

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -8,16 +8,35 @@ def current_user? user
...
+    cookies.permanent[:remember_token] = user.remember_token
   end
 
   def logged_in?
     !current_user.nil?
   end
 
+  def forget user
+    user.forget
+    cookies.delete(:user_id)
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: remove `()`

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -8,16 +8,35 @@ def current_user? user
...
+    cookies.delete(:remember_token)
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: remove `()`

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -8,16 +8,35 @@ def current_user? user
   end
 
   def current_user
-    user_id = session[:user_id]
-    @current_user ||= User.find_by(id: user_id)
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: remove `()`

File: app/helpers/sessions_helper.rb [hunk 4]
```diff
@@ -8,16 +8,35 @@ def current_user? user
   end
 
   def current_user
-    user_id = session[:user_id]
-    @current_user ||= User.find_by(id: user_id)
+    if (user_id = session[:user_id])
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: remove `()`

File: app/helpers/sessions_helper.rb [hunk 5]
```diff
@@ -8,16 +8,35 @@ def current_user? user
...
+      @current_user ||= User.find_by(id: user_id)
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: remove `()`

File: app/helpers/sessions_helper.rb [hunk 6]
```diff
@@ -8,16 +8,35 @@ def current_user? user
...
+    elsif (user_id = cookies.signed[:user_id])
```
Comment (dieunb) [inline app/helpers/sessions_helper.rb]: remove `()`

File: app/models/user.rb [hunk 7]
```diff
@@ -15,4 +17,24 @@ class User < ActiveRecord::Base
...
+    BCrypt::Password.create string, cost: cost
+  end
+
+  def User.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
```
Comment (dieunb) [inline app/models/user.rb]: use `update_attributes` and compare with `update_attribute`

**Pull request merged**